File: files/CH-053.py
import pandas as pd

input = pd.read_excel("CH-053 OEIS Sequence.xlsx", usecols="B:C", skiprows=1, nrows=10)
test = pd.read_excel("CH-053 OEIS Sequence.xlsx", usecols="G:G", skiprows=1)

range = pd.DataFrame({"number": range(101)})

def are_digits_alphabetical(number):
    digits = list(str(number))
    replaced = [input.loc[input["Number"] == int(digit), "Text"].values[0] for digit in digits]
    return replaced == sorted(replaced)

range["is_alphabetical"] = range["number"].apply(are_digits_alphabetical)
range = range[range["is_alphabetical"]].reset_index(drop=True)
range = range["number"]

print(range.equals(test["Customer"])) # True

File: files/000-099/053/CH-053.R
library(tidyverse)
library(readxl)

input = read_excel("files/CH-053 OEIS Sequence.xlsx", range = "B2:C12")
test  = read_excel("files/CH-053 OEIS Sequence.xlsx", range = "G2:G67")

range = data.frame(number = 0:100 %>% as.numeric())

are_digits_alphabetical = function(number) {
  digits = as.character(number) %>% strsplit("") %>% unlist()
  replaced = map_chr(digits, ~input$Text[match(.x, input$Number)] %>% as.character())
  all(replaced == sort(replaced))
}

result = range %>% 
  mutate(alphabetical = map_lgl(number, are_digits_alphabetical)) %>%
  filter(alphabetical) %>%
  select(number)

identical(result$number, test$Customer)
#> [1] TRUE 
